### PR TITLE
Remove insecure endpoints - Closes #457

### LINKED ISSUE #457: Remove all insecure POST API endpoints
We should remove all API endpoints receiving a passphrase/secret over the network. Instead we should rely on the signing transactions locally using `lisk-js`, and then post them onto the network via some rewritten: `POST /api/transactions` endpoint, or in case of mult-signatures: `POST /api/multisignatures/sign`.

**Endpoints that need to be first rewritten, to facilitate functional testing of public API:**

- [x] Rewrite `PUT /api/transactions` -> `POST /api/transactions`
- [x] Rewrite `POST /api/multisignatures/sign` -> `POST  /api/signatures`

Note: These API endpoints should behave in same way as:

- `POST /peer/transactions`
- `POST /peer/signatures`

Which should mean this code can be shared between p2p and public APIs.

These end points should recieve locally signed transactions or multi-signatures from `lisk-js`, and process them, without entering the p2p stack e.g. passing any peers data schema checks or being interpreted as a peer communication.

Functional tests previously reliant on these endpoints will need to be adjusted to use the `lisk-js` driven method of local signing and then posting to these revised API endpoints.

**Endpoints to be removed are as follows:**

- [x] Remove `POST /api/accounts/open`
- [x] Remove `POST /api/accounts/generatePublicKey`
- [x] Remove `PUT /api/accounts/delegates`
- [x] Remove `PUT /api/delegates`
- [x] Remove `PUT /api/signatures`
- [x] Remove `PUT /api/dapps`
- [x] Remove `PUT /api/multisignatures`

### PR DESCRIPTION
**Pull request by MaciejBaj**
remove insecure endpoints and move corresponding tests:
- move public signatures tests to the transport layer
- rename transport layer tests using `transport` prefix
- use lisk-js in functional tests to fill removed lisk functionalities
- remove create functions from all transaction types

closes #457

### CODE REVIEW

File: modules/transport.js [hunk 1]
```diff
@@ -240,7 +242,11 @@ __private.receiveTransaction = function (transaction, peer, extraLogMessage, cb)
 	}
 
 	library.balancesSequence.add(function (cb) {
-		library.logger.debug('Received transaction ' + transaction.id + ' from peer ' + library.logic.peers.peersManager.getAddress(peer.nonce));
+		if (!peer) {
+			library.logger.debug('Received transaction ' + transaction.id + ' from public client');
+		} else {
+			library.logger.debug('Received transaction ' + transaction.id + ' from peer ' + library.logic.peers.peersManager.getAddress(peer.nonce));
+		}
```
Comment (Isabello) [inline modules/transport.js]: I like this quite a bit!

File: test/api/transport/transport.transactions.votes.js [hunk 2]
```diff
@@ -77,6 +73,20 @@ describe('postTransactions', function () {
 			secret: node.gAccount.password,
 			amount: 100000000000,
 			recipientId: account.address
+		}, function (err, res) {
+			node.expect(err).to.be.null;
+			node.onNewBlock(done);
```
Comment (Isabello) [inline test/api/transport/transport.transactions.votes.js]: Could be fragile

File: api/http/blocks.js [hunk 3]
```diff
@@ -7,18 +7,18 @@ var httpApi = require('../../helpers/httpApi');
...
- * 	- get	/getEpoch
- * 	- get	/getHeight
- * 	- get	/getNethash
- * 	- get	/getFee
- * 	- get	/getFees
- * 	- get	/getMilestone
- * 	- get	/getReward
- * 	- get	/getSupply
- * 	- get	/getStatus
+	  - get		/
```
Comment (Isabello) [inline api/http/blocks.js]: this looks like crap. is there a reason we have it like this?

File: api/http/delegates.js [hunk 4]
```diff
@@ -7,22 +7,21 @@ var httpApi = require('../../helpers/httpApi');
...
+		- get 	/fee
+		- get 	/forging/getForgedByAccount
+		- get	  /getNextForgers
  * - Private API:
- * 	- post 	/forging/enable
- * 	- post 	/forging/disable
- * 	- get 	/forging/status
+		- post 	/forging/enable
+	 	- post 	/forging/disable
+	  - get 	/forging/status
```
Comment (Isabello) [inline api/http/delegates.js]: Seriously whats going on with this indention?

File: modules/multisignatures.js [hunk 5]
```diff
@@ -63,6 +63,9 @@ function Multisignatures (cb, scope) {
  * @todo test function!.
  */
 Multisignatures.prototype.processSignature = function (tx, cb) {
+	if (!tx) {
+		return setImmediate(cb, 'Cannot process empty transaction.');
```
Comment (Isabello) [inline modules/multisignatures.js]: What does this mean? Can we give a more explicit error than this?

File: modules/transactions.js [hunk 6]
```diff
@@ -580,6 +580,7 @@ Transactions.prototype.onBind = function (scope) {
 	modules = {
 		accounts: scope.accounts,
 		transactions: scope.transactions,
+		transport: scope.transport,
```
Comment (Isabello) [inline modules/transactions.js]: comma is going to break grunt i think.

File: test/api/transport/transport.js [hunk 7]
```diff
@@ -197,22 +376,20 @@ describe('RPC', function () {
 
 		it('should should work ok with asking for a list multiple times', function (done) {
 
-			var count = 0;
+			var successfullAsks = 0;
```
Comment (Isabello) [inline test/api/transport/transport.js]: spelling. `successfulAsks`

File: test/api/transport/transport.transactions.main.js [hunk 8]
```diff
@@ -0,0 +1,258 @@
+'use strict';
+
+var crypto = require('crypto');
+var node = require('../../node.js');
+var _ = require('lodash');
+var ws = require('../../common/wsCommunication.js');
+var http = require('../../common/httpCommunication.js');
```
Comment (Isabello) [inline test/api/transport/transport.transactions.main.js]: reorder these, modules first, then file includes

File: api/http/delegates.js (hunk 4 above)
Comment (MaciejBaj) [inline api/http/delegates.js]: @diego-G

File: api/http/blocks.js (hunk 3 above)
Comment (MaciejBaj) [inline api/http/blocks.js]: @diego-G

File: modules/multisignatures.js (hunk 5 above)
Comment (MaciejBaj) [inline modules/multisignatures.js]: that someone just send the empty signature like: `signature: null`

File: test/api/transport/transport.js (hunk 7 above)
Comment (MaciejBaj) [inline test/api/transport/transport.js]: thanks

File: api/http/multisignatures.js [hunk 9]
```diff
@@ -26,8 +24,6 @@ function MultisignaturesHttpApi (mutlisignaturesModule, app) {
 
 	router.map(mutlisignaturesModule.shared, {
```
Comment (MaciejBaj) [inline api/http/multisignatures.js]: ok, I will correct it

**Comment by Isabello** [comment]
Now lets get websockets un so we can merge this!!

File: logic/signature.js [hunk 10]
```diff
@@ -33,22 +33,6 @@ Signature.prototype.bind = function (accounts) {
...
-	trs.amount = 0;
-	trs.asset.signature = {
-		publicKey: data.secondKeypair.publicKey.toString('hex')
-	};
-
-	return trs;
-};
-
-/**
  * Obtains constant fee secondsignature.
```
Comment (Isabello) [inline logic/signature.js]: secondSignature

File: modules/multisignatures.js [hunk 11]
```diff
@@ -63,6 +63,9 @@ function Multisignatures (cb, scope) {
  * @todo test function!.
  */
 Multisignatures.prototype.processSignature = function (tx, cb) {
+	if (!tx) {
+		return setImmediate(cb, 'Cannot process an empty signature.');
```
Comment (Isabello) [inline modules/multisignatures.js]: something like this. Not sure on the wording myself.

File: logic/signature.js (hunk 10 above)
Comment (diego-G) [inline logic/signature.js]: This implies several changes coming from the helper constants. @Isabello Shall we follow the same pattern to `multiSignature`?

Comment (Isabello) [inline logic/signature.js]: Ah damn. No lets not change this until later. Please log an issue about standardizing it to be camelCase if you don't mind.

**Pull request merged**